File: 8-9.java
import java.util.*;

public class Main {
    public static String solution(int n, int k) {
        String a = "";
        while (n > 0) {
            int d = n % k;
            n = n / k;
            a += Integer.toString(d);
        }
        a = new StringBuilder(a).reverse().toString();
        a += "0";
        long c = 0;
        int pos = 0;
        int prev_pos = 0;
        while ((pos = a.indexOf('0', pos)) != -1) {
            String b = a.substring(prev_pos, pos);
            if (!b.isEmpty()) {
                c += Long.parseLong(b);
            }
            pos++;
            prev_pos = pos;
        }
        String ret = "";
        while (c > 0) {
            int d = (int) (c % k);
            c = c / k;
            ret += Integer.toString(d);
        }
        ret = new StringBuilder(ret).reverse().toString();
        return ret;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = scanner.nextInt();
        System.out.println(solution(n, k));
    }
}

File: 8-9.cpp
#include <bits/stdc++.h>
using namespace std;

string solution(int n, int k) {
    string a = "";
    while (n > 0) {
        int d = n % k;
        n = n / k;
        a += to_string(d);
    }
    reverse(a.begin(), a.end());
    a += "0";
    long long c = 0;
    size_t pos = 0;
    size_t prev_pos = 0;
    while ((pos = a.find('0', pos)) != string::npos) {
        string b = a.substr(prev_pos, pos - prev_pos);
        if (!b.empty()) {
            c += stoll(b);
        }
        pos++;
        prev_pos = pos;
    }
    string ret = "";
    while (c > 0) {
        int d = c % k;
        c = c / k;
        ret += to_string(d);
    }
    reverse(ret.begin(), ret.end());
    return ret;
}

int main() {
    int n, k;
    cin >> n >> k;
    cout << solution(n, k) << endl;
    return 0;
}